openstack/compute/v2/servers/results.go

```diff
@@ -20,8 +22,26 @@ func (r serverResult) Extract() (*Server, error) {
 		Server Server `mapstructure:"server"`
 	}
 
-	err := mapstructure.Decode(r.Body, &response)
-	return &response.Server, err
+	config := &mapstructure.DecoderConfig{
+		DecodeHook: func(from reflect.Kind, to reflect.Kind, data interface{}) (interface{}, error) {
+			if (from == reflect.String) && (to == reflect.Map) {
+				return map[string]interface{}{}, nil
+			}
+			return data, nil
+		},
+		Result: &response,
+	}
```

Comment:
I considered that, but the `Result` field has different types for each of the functions.